pkg/ingester/wal.go

```diff
@@ -241,6 +241,24 @@ func (w *walWrapper) performCheckpoint() (err error) {
 		os.RemoveAll(checkpointDirTemp)
 	}()
 
+	// Count number of series - we'll use this to rate limit checkpoints.
+	numSeries := 0
+	for _, state := range w.getUserStates() {
```

Comment:
`w.getUserStates()` does a copy of the map and it's also called below within the same file. You may just call it once and then reuse it.

Reply:
Right, modified